addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/model/GsonUtil.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.tplinksmarthome.internal.model;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.annotations.Expose;
+
+/**
+ * Util class to create a {@link Gson} object.
+ *
+ * @author Hilbrand Bouwkamp - Initial contribution
+ */
+@NonNullByDefault
+@SuppressWarnings("null")
+public final class GsonUtil {
+
+    private GsonUtil() {
+        // Util class
+    }
+
+    /**
+     * Creates a new {@link Gson} object.
+     *
+     * @return new {@link Gson} object.
+     */
+    public static Gson createGson() {
+        return new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create();
+    }
+
+    /**
+     * Creates a new {@link Gson} object that uses the {@link Expose} annotation..
+     *
+     * @return new {@link Gson} object.
+     */
+    public static Gson createGsonWithExpose() {
```

Comment:
Why do you need two, is one not enough? I would guess that you fiddle with the configuration to use just one. Also the number of calls to this util is limited should it live on its own? I'm not such a fan of util classes.

Also for future extension by others it would be clearer to have one style. What is the use case where you need to exclude fields?

PS: i'm a jackson user myself and shocked to see that they make this so hard in GSON :-(

Reply:
Yes using expose is not handled well. There is a change request if I remember correctly or something improved in later versions. I use two because some commands data objects are used both the send and receive. Because the receive has more data I used the Expose to not send all unused fields to the device. Possibly technical it won't make a difference. But it's just to have a clean implementation. The second without expose is because some received data objects are relatively large and thus would require Expose annotation on each field. That was a bit too much imho